test/unit/org/apache/cassandra/db/filter/ColumnFilterTest.java

```diff
@@ -416,6 +470,16 @@ else if ("3.11".equals(clusterMinVersion))
             assertCellFetchedQueried(true, false, filter, v2, path0, path2, path3, path4);
             assertCellFetchedQueried(true, false, filter, s2, path0, path1, path2, path3, path4);
         }
+        else if (returnStaticContentOnPartitionWithNoRows && "4.0".equals(clusterMinVersion))
+        {
+            assertEquals("*/[v2[1]]", filter.toString());
+            assertEquals("v2[1]", filter.toCQLString());
+            assertFetchedQueried(true, false, filter, v1);
+            assertFetchedQueried(true, false, filter, s1, s2);
+            assertCellFetchedQueried(true, true, filter, v2, path1);
+            assertCellFetchedQueried(true, false, filter, v2, path0, path2, path3, path4);
+            assertCellFetchedQueried(false, false, filter, s2, path0, path1, path2, path3, path4);
```

Comment:
I guess we should expect `fetches=true` here since all statics are fetched

Reply:
In general, isn't the case `4.0` and `returnStaticContentOnPartitionWithNoRows` identical to `3.11` ?

Reply:
Good point! I will merge the `if` branches